### PR TITLE
Finish converting migration guides to MDX

### PR DESCRIPTION
**Pull request by Eric-Arellano**
Closes https://github.com/Qiskit/documentation/issues/213.

These changes are all manual. I looked over the rendered site and MDX files to see issues like `[My link]` without the corresponding `()` for the link itself.  But for tables, I used the script from https://github.com/Qiskit/documentation/pull/296.

### CODE REVIEW

File: docs/api/migration-guides/_toc.json [hunk 1]
```diff
@@ -17,6 +17,23 @@
             "url": "/api/migration-guides/qiskit-runtime-examples"
           }
         ]
+      },
+      {
+        "title": "Qiskit 0.44 changes",
```
Comment (Eric-Arellano) [inline docs/api/migration-guides/_toc.json:22]: @ElePT and I talked this over. We couldn't think of a better name.

File: docs/api/migration-guides/qiskit-opflow-module.mdx [hunk 2]
```diff
@@ -357,22 +357,11 @@ to initialize it.
 <Admonition type="tip">
   Interpreting [`qiskit.opflow.primitive_ops.PrimitiveOp`](../qiskit/qiskit.opflow.primitive_ops.PrimitiveOp) as a factory class:
 
-  ```{eval-rst}
```
Comment (Eric-Arellano) [inline docs/api/migration-guides/qiskit-opflow-module.mdx:360]: Tables were converted via the script from https://github.com/Qiskit/documentation/pull/296

File: docs/api/migration-guides/qiskit-opflow-module.mdx [hunk 3]
```diff
@@ -1260,42 +1236,23 @@ qfi = QFI(qgt)
   Here is a quick guide for migrating the most common gradient settings. Please note that all new gradient
   imports follow the format:
 
-  > ```python
-  > from qiskit.algorithms.gradients import MethodPrimitiveGradient, QFI
-  > ```
```
Comment (Eric-Arellano) [inline docs/api/migration-guides/qiskit-opflow-module.mdx:1265]: This code block was being indented with `>`. I removed that.

File: docs/api/migration-guides/qiskit-quantum-instance.mdx [hunk 4]
```diff
@@ -34,9 +34,9 @@ The remainder of this guide will focus on the [`qiskit.utils.QuantumInstance.exe
```
Comment (Eric-Arellano) [inline docs/api/migration-guides/qiskit-quantum-instance.mdx:34]: The diff is pretty bad in this file. I will call out where I made changes.

File: docs/api/migration-guides/qiskit-quantum-instance.mdx [hunk 5]
```diff
@@ -144,481 +144,463 @@ primitives **expose a similar setting through their interface**:
 
 ## Code examples
 
-```{eval-rst}
-.. dropdown:: Example 1: Circuit Sampling with Local Simulation
-    :animate: fade-in-slide-down
+### Example 1: Circuit Sampling with Local Simulation
```
Comment (Eric-Arellano) [inline docs/api/migration-guides/qiskit-quantum-instance.mdx:147]: There are 4 examples that were dropdowns. For each of them, I added a header and dedented all the content. For the content itself, I replaced two backticks with a single backtick. I also converted code blocks to Markdown.

File: docs/api/migration-guides/qiskit-quantum-instance.mdx [hunk 6]
```diff
@@ -144,481 +144,463 @@ primitives **expose a similar setting through their interface**:
...
 
-        -0.04687500000000008
+```text
+Quasi-dists: [{3: 1.0}]
+Result: SamplerResult(quasi_dists=[{3: 1.0}], metadata=[{'shots': 200, 'simulator_metadata': {'parallel_state_update': 16, 'parallel_shots': 1, 'sample_measure_time': 9.016e-05, 'noise': 'ideal', 'batched_shots_optimization': False, 'remapped_qubits': False, 'device': 'CPU', 'active_input_qubits': [0, 1], 'measure_sampling': True, 'num_clbits': 2, 'input_qubit_map': [[1, 1], [0, 0]], 'num_qubits': 2, 'method': 'statevector', 'fusion': {'applied': False, 'max_fused_qubits': 5, 'threshold': 14, 'enabled': True}}}])
+Binary quasi-dist:  {'11': 1.0}
+```
 
-    **Using Primitives**
+### Example 2: Expectation Value Calculation with Local Noisy Simulation
```
Comment (Eric-Arellano) [inline docs/api/migration-guides/qiskit-quantum-instance.mdx:261]: Example 2

File: docs/api/migration-guides/qiskit-quantum-instance.mdx [hunk 7]
```diff
@@ -144,481 +144,463 @@ primitives **expose a similar setting through their interface**:
...
-        circuit.x(0)
-        circuit.x(1)
-        circuit.measure_all()
+```python
+[-0.04101562]
+```
 
-        provider = IBMProvider()
-        backend = provider.get_backend("ibmq_montreal")
+### Example 3: Circuit Sampling on IBM Backend with Error Mitigation
```
Comment (Eric-Arellano) [inline docs/api/migration-guides/qiskit-quantum-instance.mdx:356]: Example 3

File: docs/api/migration-guides/qiskit-quantum-instance.mdx [hunk 8]
```diff
@@ -144,481 +144,463 @@ primitives **expose a similar setting through their interface**:
...
-```{eval-rst}
-.. dropdown:: Example 4: Circuit Sampling with Custom Bound and Unbound Pass Managers
-    :animate: fade-in-slide-down
+```text
+Quasi dists: [{2: 0.0008492371522941081, 3: 0.9968874384378738, 0: -0.0003921227905920063,
+            1: 0.002655447200424097}]
+```
 
-    The management of transpilation is different between the ``QuantumInstance`` and the Primitives.
+### Example 4: Circuit Sampling with Custom Bound and Unbound Pass Managers
```
Comment (Eric-Arellano) [inline docs/api/migration-guides/qiskit-quantum-instance.mdx:429]: Example 4

File: docs/api/migration-guides/qiskit-quantum-instance.mdx [hunk 9]
```diff
@@ -144,481 +144,463 @@ primitives **expose a similar setting through their interface**:
...
-
-    **b. Using the Aer Primitives**
-
-    Aer simulation following the statevector method. This would be the closer replacement of the
-    [`qiskit.utils.QuantumInstance`](../qiskit/qiskit.utils.QuantumInstance)
-    example, as they are both accessing the same simulator. For this reason, the output metadata is
-    closer to the Quantum Instance's output. Please note that
-    the resulting quasi-probability distribution does not use bitstrings but **integers** to identify the states.
-
-    .. note::
```
Comment (Eric-Arellano) [inline docs/api/migration-guides/qiskit-quantum-instance.mdx:227]: This was converted to an `<Admonition type="note">`

File: docs/api/migration-guides/qiskit-quantum-instance.mdx [hunk 10]
```diff
@@ -144,481 +144,463 @@ primitives **expose a similar setting through their interface**:
...
+    not have any free parameters.
 
-    * You cannot explicitly access their transpilation routine.
-    * The mechanism to apply custom transpilation passes to the Aer, Runtime and Backend primitives is to pre-transpile
-      locally and set ``skip_transpilation=True`` in the corresponding primitive.
-    * The only primitives that currently accept a custom **bound** transpiler pass manager are instances of [`qiskit.primitives.BackendSampler`](../qiskit/qiskit.primitives.BackendSampler) or [`qiskit.primitives.BackendEstimator`](../qiskit/qiskit.primitives.BackendEstimator).
-      If a ``bound_pass_manager`` is defined, the ``skip_transpilation=True`` option will **not** skip this bound pass.
+On the other hand, when using the primitives:
 
-    .. attention::
```
Comment (Eric-Arellano) [inline docs/api/migration-guides/qiskit-quantum-instance.mdx:465]: This was converted to `<Admonition type="caution">`

File: docs/api/migration-guides/qiskit-quantum-instance.mdx [hunk 11]
```diff
@@ -144,481 +144,463 @@ primitives **expose a similar setting through their interface**:
...
+sampler = CircuitSampler(quantum_instance)
+
+# Run
+quasi_dists = sampler.convert(StateFn(qc), params=param_dict).sample()
+print("Quasi-dists: ", quasi_dists)
+```
 
-    ..
-        In opflow, the ansatz would always have the basis change and measurement gates added before transpilation,
-        so if the circuit ended up on more qubits it did not matter.
```
Comment (Eric-Arellano) [inline docs/api/migration-guides/qiskit-quantum-instance.mdx:474]: This was removed. It's a comment and not rendered to the user.

**Review by Eric-Arellano — COMMENTED** [review]
To review, you can preview:

* https://qiskit-docs-preview-pr-304.1799mxdls7qz.us-south.codeengine.appdomain.cloud/api/migration-guides/qiskit-algorithms-module
* https://qiskit-docs-preview-pr-304.1799mxdls7qz.us-south.codeengine.appdomain.cloud/api/migration-guides/qiskit-opflow-module
* https://qiskit-docs-preview-pr-304.1799mxdls7qz.us-south.codeengine.appdomain.cloud/api/migration-guides/qiskit-quantum-instance

**Review by kevinsung — APPROVED** [review]

**Pull request merged**